staging/src/k8s.io/apiserver/pkg/util/version/version.go

```diff
@@ -0,0 +1,175 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"fmt"
+	"strings"
+	"sync/atomic"
+
+	"github.com/spf13/pflag"
+	"k8s.io/apimachinery/pkg/util/version"
+	genericfeatures "k8s.io/apiserver/pkg/features"
+	utilfeature "k8s.io/apiserver/pkg/util/feature"
+	"k8s.io/component-base/featuregate"
+	baseversion "k8s.io/component-base/version"
+)
+
+var Effective MutableEffectiveVersions = newEffectiveVersion()
```

Comment:
Do all components (kube-scheduler, kube-controller, etc) that will make use of compatibility version do so by using a k8s.io/apiserver package? I thought this would live somewhere like k8s.io/component-base, where k8s.io/component-base/version{/verflag} currently lives.

A global accessor doesn't necessarily bother me (k8s.io/component-base/version is global, after all, and a given binary *does* have a distinct version), but things registering hard-coded versions into the global variable seems off. What are we expecting to happen when something based on k8s.io/component-base builds, stamps in its own version, and sets up a client-go client that registers a bunch of Kubernetes APIs into the client scheme? It seems *very* strange to have an EffectiveVersion global that would then have Kubernetes minor version data in it.